ts/utils/input.ts

```diff
@@ -37,3 +40,85 @@ export type CreditCardExpirationYear = t.TypeOf<
  */
 export const CreditCardCVC = PatternString("^[0-9]{3,4}$");
 export type CreditCardCVC = t.TypeOf<typeof CreditCardCVC>;
+
+export const INITIAL_CARD_FORM_STATE: CreditCardState = {
+  pan: none,
+  expirationDate: none,
+  securityCode: none,
+  holder: none
+};
+
+export const isValidPan = (pan: Option<string>) =>
+  pan.map(pan => CreditCardPan.is(pan)).toUndefined();
+
+export const isValidExpirationDate = (expirationDate: Option<string>) =>
+  expirationDate
+    .map(expirationDate => {
+      const [expirationMonth, expirationYear] = expirationDate.split("/");
+      return (
+        CreditCardExpirationMonth.is(expirationMonth) &&
+        CreditCardExpirationYear.is(expirationYear) &&
+        !isExpired(Number(expirationMonth), Number(expirationYear))
+      );
+    })
+    .toUndefined();
+
+export const isValidSecurityCode = (securityCode: Option<string>) =>
+  securityCode
+    .map(securityCode => CreditCardCVC.is(securityCode))
+    .toUndefined();
+
+export type CreditCardState = Readonly<{
+  pan: Option<string>;
+  expirationDate: Option<string>;
+  securityCode: Option<string>;
+  holder: Option<string>;
+}>;
+
+export type CreditCardStateKeys =
+  | "pan"
+  | "expirationDate"
+  | "securityCode"
+  | "holder";
+
+export function mergeCardValues(state: CreditCardState): Option<CreditCard> {
```

Comment:
Could you leave a comment describing what this function does?
Could `getCreditCardFromState` be a more expressive name?

Reply:
You're right, no real need to rename it since it refers to a "state".